libparsec/crates/types/src/local_device.rs

```diff
@@ -60,13 +62,29 @@ impl LocalDevice {
             signing_key: signing_key.unwrap_or_else(SigningKey::generate),
             private_key: private_key.unwrap_or_else(PrivateKey::generate),
             initial_profile,
-            user_realm_id: VlobID::default(),
-            user_realm_key: SecretKey::generate(),
+            user_realm_id: user_realm_id.unwrap_or_default(),
+            user_realm_key: user_real_key.unwrap_or_else(SecretKey::generate),
             local_symkey: SecretKey::generate(),
```

Comment:
shouldn't`local_symkey` be corrected too ?

Reply:
And if we avoid using generate_new_device after #8790 it won't be needed.

Reply:
Actually that fix is not needed anymore

Reply:
As you say, we should deal with this question in #8790